src/pages/Admin/Venue/Users/VenueUsersReport.tsx

```diff
@@ -0,0 +1,166 @@
+import React, { useEffect, useCallback, useState } from "react";
+import { useWorldUsers } from "hooks/users";
+import "./VenueUsersReport.scss";
+import Table from "react-bootstrap/Table";
+import Form from "react-bootstrap/Form";
+import { downloadGeneratedCSVFile } from "utils/csv";
+import { User } from "types/User";
+import { WithId } from "utils/id";
+
+const VenueUsersReport: React.FC = () => {
+  const { worldUsers } = useWorldUsers();
+
+  const [headers, setHeaders] = useState<Array<string>>([]);
+  const [headersSelected, setHeadersSelected] = useState<Array<string>>([]);
+  const [customFieldValue, setCustomFieldValue] = useState("");
+
+  const userIsComplete = useCallback((user: WithId<User>) => {
+    return user.partyName && user.pictureUrl;
+  }, []);
+
+  const getUsersHeaders = useCallback(() => {
+    let index = 0;
+    while (index < worldUsers.length) {
+      if (userIsComplete(worldUsers[index])) {
+        setHeaders(Object.keys(worldUsers[index]));
+        return;
+      } else {
+        index++;
+      }
+    }
+  }, [worldUsers, userIsComplete]);
+
+  useEffect(() => {
+    getUsersHeaders();
+  }, [worldUsers, getUsersHeaders]);
+
+  const renderPageHeader = () => {
+    return <h4>Users Reports</h4>;
+  };
+
+  const renderUserTable = () => {
+    return (
+      <Table striped bordered hover>
+        <thead>
+          <tr>
+            <th>#</th>
+            <th>Party Name</th>
+            <th>Full Name</th>
+            <th>Department</th>
+          </tr>
+        </thead>
+        <tbody>
+          {/* when I try to use WithId<User> as realName are not added in it, it shows erro, so that's why I need any */}
+          {/* eslint-disable-next-line*/}
+          {worldUsers.map((user: any, index) => {
+            return (
+              userIsComplete(user) && (
+                <tr key={index}>
```

Comment:
Never `key` things with `index`. It's an anti pattern that causes hard to identify rendering issues. `key` must always be a stable/unique identifier that uniquely identifies the record it is relating to. I believe there is an old comment/thread in `#tech` on slack you can search for that goes into this in more detail.
```suggestion
                <tr key={user.id}>
```

Reply:
fixed in 12e5c07fea64a78aced31d706faf59c5e523c733